### ISSUE TITLE
Can't type in steam overlay

### ISSUE THREAD

**Issue by harleypig** [ValveSoftware/steam-for-linux#6890]
#### Your system information

* Steam client version (build number or date):
Build: Jan 18 2020, at 03:34:03
Steam API: v020
Steam package versions: 1579321278

* Distribution (e.g. Ubuntu):
Arch Linux

* Opted into Steam client beta?: [Yes/No]
Yes

* Have you checked for system updates?: [Yes/No]
I update my system weekly. Steam says it's up to date.

#### Please describe your issue in as much detail as possible:
In Steam client, overlay and CS:GO, I am unable to type in any input field. In CS:GO I can use the keyboard for in game commands (AWSDQF123, F12 all work) and pressing Y brings up chat, but I can't type in the chat window.

#### Steps for reproducing this issue:
I have no idea.

**Comment by harleypig** [comment]
I forgot to mention that I can type in Creeper World with no problems, as well as there are no problems in any other app or console on my system.

I have deleted ~/.local and ~/.config and reinstalled. Same thing.

**Comment by TTimo** [comment]
Please provide the output of Help > System Information, with the `Steam Linux Runtime Information` section.

**Comment by harleypig** [comment]
```
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:         Intel(R) Core(TM) i7-4820K CPU @ 3.70GHz
    CPU Family:  0x6
    CPU Model:  0x3e
    CPU Stepping:  0x4
    CPU Type:  0x0
    Speed:  3900 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    AVX2:  Unsupported
    AVX512F:  Unsupported
    AVX512PF:  Unsupported
    AVX512ER:  Unsupported
    AVX512CD:  Unsupported
    AVX512VNNI:  Unsupported
    SHA:  Unsupported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "Arch Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  5.5.3-arch1-1
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12007000
    X Window Manager:  GNOME Shell
    Steam Runtime Version:  steam-runtime_0.20200204.0

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 970/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 440.59
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x10de
    DeviceID:  0x13c2
    Revision Not Detected
    Number of Monitors:  3
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 5760 x 1080
    Primary Display Size: 20.51" x 11.54" (23.50" diag)
                                            52.1cm x 29.3cm (59.7cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 4096 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC892

Memory:
    RAM:  64262 Mb

VR Hardware:
    VR Headset: None detected

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  3657230 Mb
    Largest Free Hard Disk Block:  1954157 Mb

Steam Linux Runtime Information:
{
  "can-write-uinput" : true,
  "steam-installation" : {
    "path" : "/home/harleypig/.local/share/Steam",
    "data_path" : "/home/harleypig/.local/share/Steam",
    "issues" : [
    ]
  },
  "runtime" : {
    "path" : "/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime",
    "version" : "0.20200204.0",
    "issues" : [
    ],
    "pinned_libs_32" : {
      "list" : [
        "176557170      4 drwxr-xr-x   2 harleypig harleypig     4096 Feb 12 00:03 pinned_libs_32",
        "176557176      4 -rw-r--r--   1 harleypig harleypig       70 Feb 12 00:03 pinned_libs_32/system_libgtk-x11-2.0.so.0",
        "176557172      4 lrwxrwxrwx   1 harleypig harleypig      109 Feb 12 00:03 pinned_libs_32/libdbusmenu-gtk.so.4 -> /home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libdbusmenu-gtk.so.4.0.13",
        "176557175      4 lrwxrwxrwx   1 harleypig harleypig      111 Feb 12 00:03 pinned_libs_32/libgtk-x11-2.0.so.0 -> /home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libgtk-x11-2.0.so.0.2400.10",
        "176557178      4 -rw-r--r--   1 harleypig harleypig       71 Feb 12 00:03 pinned_libs_32/system_libdbusmenu-glib.so.4",
        "176557174      0 -rw-r--r--   1 harleypig harleypig        0 Feb 12 00:03 pinned_libs_32/has_pins",
        "176557177      4 lrwxrwxrwx   1 harleypig harleypig      110 Feb 12 00:03 pinned_libs_32/libdbusmenu-glib.so.4 -> /home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libdbusmenu-glib.so.4.0.13",
        "176557173      4 -rw-r--r--   1 harleypig harleypig       69 Feb 12 00:03 pinned_libs_32/system_libdbusmenu-gtk.so.4"
      ]
    },
    "pinned_libs_64" : {
      "list" : [
        "176557171      4 drwxr-xr-x   2 harleypig harleypig     4096 Feb 12 00:03 pinned_libs_64",
        "176557179      4 lrwxrwxrwx   1 harleypig harleypig      111 Feb 12 00:03 pinned_libs_64/libdbusmenu-gtk.so.4 -> /home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libdbusmenu-gtk.so.4.0.13",
        "176557182      4 lrwxrwxrwx   1 harleypig harleypig      102 Feb 12 00:03 pinned_libs_64/libcurl.so.4 -> /home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libcurl.so.4.2.0",
        "176557184      4 lrwxrwxrwx   1 harleypig harleypig      105 Feb 12 00:03 pinned_libs_64/libGLU.so.1 -> /home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libGLU.so.1.3.08004",
        "176557189      4 -rw-r--r--   1 harleypig harleypig       49 Feb 12 00:03 pinned_libs_64/system_libjack.so.0",
        "176557185      4 -rw-r--r--   1 harleypig harleypig       46 Feb 12 00:03 pinned_libs_64/system_libGLU.so.1",
        "176557183      4 -rw-r--r--   1 harleypig harleypig       48 Feb 12 00:03 pinned_libs_64/system_libcurl.so.4",
        "176557187      4 -rw-r--r--   1 harleypig harleypig       67 Feb 12 00:03 pinned_libs_64/system_libdbusmenu-glib.so.4",
        "176557181      0 -rw-r--r--   1 harleypig harleypig        0 Feb 12 00:03 pinned_libs_64/has_pins",
        "176557190      0 lrwxrwxrwx   1 harleypig harleypig       12 Feb 12 00:03 pinned_libs_64/libcurl.so.3 -> libcurl.so.4",
        "176557186      4 lrwxrwxrwx   1 harleypig harleypig      112 Feb 12 00:03 pinned_libs_64/libdbusmenu-glib.so.4 -> /home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libdbusmenu-glib.so.4.0.13",
        "176557188      4 lrwxrwxrwx   1 harleypig harleypig      102 Feb 12 00:03 pinned_libs_64/libjack.so.0 -> /home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libjack.so.0.1.0",
        "176557180      4 -rw-r--r--   1 harleypig harleypig       65 Feb 12 00:03 pinned_libs_64/system_libdbusmenu-gtk.so.4"
      ]
    }
  },
  "os-release" : {
    "id" : "arch",
    "name" : "Arch Linux",
    "pretty_name" : "Arch Linux",
    "build_id" : "rolling"
  },
  "architectures" : {
    "i386-linux-gnu" : {
      "can-run" : true,
      "library-issues-summary" : [
      ],
      "graphics-details" : {
        "x11/vulkan" : {
          "messages" : "ERROR: [Loader Message] Code 0 : /home/harleypig/.local/share/Steam/ubuntu12_64/steamoverlayvulkanlayer.so: wrong ELF class: ELFCLASS64\n",
          "renderer" : "GeForce GTX 970",
          "version" : "1.1.119 (device 10de:13c2) (driver 440.236.0)"
        },
        "glx/gl" : {
          "renderer" : "GeForce GTX 970/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.59",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "renderer" : "GeForce GTX 970/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.59",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "renderer" : "GeForce GTX 970/PCIe/SSE2",
          "version" : "OpenGL ES 3.2 NVIDIA 440.59",
          "library-vendor" : "glvnd"
        }
      },
      "dri_drivers" : [
        {
          "library_path" : "/usr/lib32/dri/i915_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/i965_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/iris_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/kms_swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/nouveau_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/nouveau_vieux_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/r200_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/r300_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/r600_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/radeon_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/radeonsi_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/virtio_gpu_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/vmwgfx_dri.so"
        }
      ],
      "va-api_drivers" : [
        {
          "library_path" : "/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/dri/dummy_drv_video.so"
        }
      ]
    },
    "x86_64-linux-gnu" : {
      "can-run" : true,
      "library-issues-summary" : [
      ],
      "graphics-details" : {
        "x11/vulkan" : {
          "messages" : "ERROR: [Loader Message] Code 0 : /home/harleypig/.local/share/Steam/ubuntu12_32/steamoverlayvulkanlayer.so: wrong ELF class: ELFCLASS32\n",
          "renderer" : "GeForce GTX 970",
          "version" : "1.1.119 (device 10de:13c2) (driver 440.236.0)"
        },
        "glx/gl" : {
          "renderer" : "GeForce GTX 970/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.59",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "renderer" : "GeForce GTX 970/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.59",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "renderer" : "GeForce GTX 970/PCIe/SSE2",
          "version" : "OpenGL ES 3.2 NVIDIA 440.59",
          "library-vendor" : "glvnd"
        }
      },
      "dri_drivers" : [
        {
          "library_path" : "/usr/lib/dri/i915_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/i965_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/iris_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/kms_swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/nouveau_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/nouveau_vieux_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/r200_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/r300_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/r600_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/radeon_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/radeonsi_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/virtio_gpu_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/vmwgfx_dri.so"
        }
      ],
      "va-api_drivers" : [
        {
          "library_path" : "/usr/lib/dri/iHD_drv_video.so"
        },
        {
          "library_path" : "/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/dri/dummy_drv_video.so"
        }
      ]
    }
  },
  "locale-issues" : [
    "c-utf8-missing",
    "i18n-supported-missing"
  ],
  "locales" : {
    "<default>" : {
      "resulting-name" : "en_US.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    },
    "C" : {
      "resulting-name" : "C",
      "charset" : "ANSI_X3.4-1968",
      "is_utf8" : false
    },
    "C.UTF-8" : {
      "error-domain" : "srt-locale-error-quark",
      "error-code" : 0,
      "error" : "No such file or directory"
    },
    "en_US.UTF-8" : {
      "resulting-name" : "en_US.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    }
  },
  "egl" : {
    "icds" : [
      {
        "json_path" : "/usr/share/glvnd/egl_vendor.d/10_nvidia.json",
        "library_path" : "libEGL_nvidia.so.0"
      },
      {
        "json_path" : "/usr/share/glvnd/egl_vendor.d/50_mesa.json",
        "library_path" : "libEGL_mesa.so.0"
      }
    ]
  },
  "vulkan" : {
    "icds" : [
      {
        "json_path" : "/usr/share/vulkan/icd.d/nvidia_icd.json",
        "library_path" : "libGLX_nvidia.so.0",
        "api_version" : "1.1.119"
      }
    ]
  }
}

The Steam Linux Runtime container tool is not installed.
```

**Comment by harleypig** [comment]
I modified csgo.sh to dump out the LD_LIBRARY_PATH, the args that called csgo and redirected the output to a logfile. I got some wrong ELFCLASS32 errors, but I have multilib installed, so I don't know if this is a bogus report:

```
LD_LIBRARY_PATH: /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin:/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/linux64:/home/harleypig/.local/share/Steam/  ubuntu12_32/steam-runtime/pinned_libs_32:/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_64:/usr/lib/libfakeroot:/usr/lib32:/usr/lib/openmpi:/usr/lib:/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/  lib/i386-linux-gnu:/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu:/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/lib/x86_64-linux-gnu:/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu:/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/lib:/home/harleypig/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib:

exec  /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/csgo_linux64 -steam

SDL video target is 'x11'
SDL failed to create GL compatibility profile (whichProfile=0!
This system supports the OpenGL extension GL_EXT_framebuffer_object.
This system supports the OpenGL extension GL_EXT_framebuffer_blit.
This system supports the OpenGL extension GL_EXT_framebuffer_multisample.
This system DOES NOT support the OpenGL extension GL_APPLE_fence.
This system supports the OpenGL extension GL_NV_fence.
This system supports the OpenGL extension GL_ARB_sync.
This system supports the OpenGL extension GL_EXT_draw_buffers2.
This system supports the OpenGL extension GL_EXT_bindable_uniform.
This system DOES NOT support the OpenGL extension GL_APPLE_flush_buffer_range.
This system supports the OpenGL extension GL_ARB_map_buffer_range.
This system supports the OpenGL extension GL_ARB_vertex_buffer_object.
This system supports the OpenGL extension GL_ARB_occlusion_query.
This system DOES NOT support the OpenGL extension GL_APPLE_texture_range.
This system DOES NOT support the OpenGL extension GL_APPLE_client_storage.
This system DOES NOT support the OpenGL extension GL_ARB_uniform_buffer.
This system supports the OpenGL extension GL_ARB_vertex_array_bgra.
This system supports the OpenGL extension GL_EXT_vertex_array_bgra.
This system supports the OpenGL extension GL_ARB_framebuffer_object.
This system DOES NOT support the OpenGL extension GL_GREMEDY_string_marker.
This system supports the OpenGL extension GL_ARB_debug_output.
This system supports the OpenGL extension GL_EXT_direct_state_access.
This system supports the OpenGL extension GL_NV_bindless_texture.
This system DOES NOT support the OpenGL extension GL_AMD_pinned_memory.
This system supports the OpenGL extension GL_EXT_framebuffer_multisample_blit_scaled.
This system supports the OpenGL extension GL_EXT_texture_sRGB_decode.
This system supports the OpenGL extension GL_NVX_gpu_memory_info.
This system DOES NOT support the OpenGL extension GL_ATI_meminfo.
This system supports the OpenGL extension GL_EXT_texture_compression_s3tc.
This system supports the OpenGL extension GL_EXT_texture_compression_dxt1.
This system DOES NOT support the OpenGL extension GL_ANGLE_texture_compression_dxt3.
This system DOES NOT support the OpenGL extension GL_ANGLE_texture_compression_dxt5.
This system supports the OpenGL extension GL_ARB_buffer_storage.
This system supports the OpenGL extension GLX_EXT_swap_control_tear.
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/engine_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/engine_client.so:    wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/filesystem_stdio_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/           filesystem_stdio_client.so: wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/inputsystem_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/                inputsystem_client.so: wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/vphysics_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/vphysics_client.   so: wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/materialsystem_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/             materialsystem_client.so: wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/datacache_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/datacache_client. so: wrong ELF class: ELFCLASS32 
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/studiorender_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/               studiorender_client.so: wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/soundemittersystem_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/         soundemittersystem_client.so: wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/vscript_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/vscript_client.so:  wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/vguimatsurface_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/             vguimatsurface_client.so: wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/vgui2_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/vgui2_client.so:      wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/shaderapidx9_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/               shaderapidx9_client.so: wrong ELF class: ELFCLASS32
 failed to dlopen /home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/localize_client.so error=/home/harleypig/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/bin/localize_client.   so: wrong ELF class: ELFCLASS32
Created D3D9 device successfully
Using breakpad minidump system 730/13746.1090.JC
Got ISteamNetworkingSockets user interfaces from standalone lib
Pending ping measurement until network config is obtained.
RelayNetWorkStatus:  avail=Attempting  config=Attempting  anyrelay=Waiting   (Attempt #1 to fetch config from https://api.steampowered.com/ISteamApps/GetSDRConfig/v1?appid=730&partner=valve)
AuthStatus (steamid:76561197969265961):  Attempting  (Requesting cert)
USRLOCAL path using Steam profile data folder:
/home/harleypig/.local/share/Steam/userdata/9000233/730/local
Did not detect any valid joysticks.
Timed out waiting for game mapping!
RESZ NOT SUPPORTED!
INTZ NOT SUPPORTED!
RESZ NOT SUPPORTED!
INTZ NOT SUPPORTED!
```

**Comment by TTimo** [comment]
The ELFCLASS32 warnings are not a problem.

> In Steam client, overlay and CS:GO

For text input not to work across all these, I suspect this is a configuration problem specific to your system and not an issue with Steam client, overlay code and CS:GO.

Does text input work in other games? Does it work in other 64 bit native games, does it work in games that are not using SDL, does it work in Proton games? That might help narrow it down.

**Comment by harleypig** [comment]
I can manually start any of the games I have and I have the same problem with CSGO, but none of the others that require or allow input.
How do I tell whether a game uses SDL or Proton or not?
In all cases, alt-tab brings up overly but I can't type in a field. alt-tab again closes the overlay.

* CS:GO - I've already reported it, I cannot type in chat window.
* Creeper World 3 - I can type in-game
* Knights of Pen and Paper - doesn't have an option for in-game keyboard entry
* Plague, Inc - I can type in-game
* Sentinel - no in-game options for keyboard input
* Wrecked - Unity, allows in-game keyboard entry

**Comment by TTimo** [comment]
Ok so looping around on this - it's only in CS:GO, you can't type in CS:GO in game chat, or in the overlay (e.g. if you bring up the overlay while playing CS:GO and try to send a chat message).

But you can type just fine in the Steam client (when not running any games) or in other games (the ones you tried at least).

Correct? (so that narrows it down a bit).

**Comment by harleypig** [comment]
I cannot type in the steam app or the overlay.

To be clear, I cannot input search terms for games in the store in the app, I cannot change my credit card information in order to purchase a game.

In CS:GO I can play the game, but I cannot type in the chat. Press Y, chat window pops up and I cannot type in the input field. I can press enter and the chat window closes.

**Comment by TTimo** [comment]
I doubt that's going to fix your problem, but have you tested the Steam beta client?

Since typing in the steam client isn't working, I'm leaning towards some problem with SDL2 and input configuration. We can probably ignore the overlay, it's going to be the same cause and it's more difficult to diagnose.

**Comment by harleypig** [comment]
I think I was already using the beta client.
Steam -> Settings -> Account -> Beta participation == Steam Beta Update
I changed it to 'No participation' and restarted the client. I was still unable to enter text into any input field in the client and in CS:GO.

**Comment by TTimo** [comment]
See if this workaround fixes it:

- Move Steam's SDL2 out of the way temporarily with:
`mv ~/.local/share/Steam/ubuntu12_32/libSDL2-2.0.so.0 ~/.local/share/Steam/ubuntu12_32/libSDL2-2.0.so.0.off`

- Run Steam with `steam -noverifyfiles`

**Comment by harleypig** [comment]
That didn't work. There is also a ubuntu12_64 directory with a file of the same name. I renamed it to 'libSDL2-2.0.so.0.off' also. Still not able to type in input fields in either steam or CS:GO.

However, I accidentally pressed 'ctl-v' in an input field in steam and what I currently had in my clipboard was pasted. I was *not* able to paste in CS:GO.

This is a bizarre issue. When I initially ran into this problem I uninstalled steam and deleted my entire .local and .cache directories and reinstalled, which did not fix the issue. I am not willing to do that again--it was a pain getting everything else back in place (even with most of it in git).

If you can give me a list of directories I should make sure are empty/gone for both steam and cs:go I will uninstall and reinstall and try again. Also, if there are any dependencies external to the steam directories, I will uninstall and reinstall them too. The pkgbuild for archlinux has the following dependencies listed.

```
depends=('bash' 'desktop-file-utils' 'hicolor-icon-theme' 'curl' 'dbus'
         'freetype2' 'gdk-pixbuf2' 'ttf-font' 'zenity' 'lsb-release' 'nss' 'usbutils')
depends_x86_64=('lib32-libgl' 'lib32-gcc-libs' 'lib32-libx11' 'lib32-libxss'
                'lib32-alsa-plugins' 'lib32-libgpg-error' 'lib32-libappindicator-gtk2'
                'lib32-libindicator-gtk2' 'lib32-libdbusmenu-glib' 'lib32-libdbusmenu-gtk2'
                'lib32-nss')
optdepends=('steam-native-runtime: steam native runtime support')
```

**Comment by harleypig** [comment]
After a bit more experimentation, if I paste a value into a field in steam I can use backspace, arrow, and shift and ctrl keys.
I can 'ctl-a' to select all the text in a field. I can 'shift-arrow' to select a sub-set of text, but I cannot type [[:alnum:]] and punctuation characters into the field.

**Comment by TTimo** [comment]
The SDL in `ubuntu12_32` is the one that would matter, since the Steam client is 32 bit. Assuming you ran with `-noverifyfiles` (otherwise Steam will just fix and replace the modified SDL2 with it's own), it sounds like it's not an SDL problem, running with your host's SDL has the same issues.

I will noddle on this some more..

**Comment by harleypig** [comment]
I can give you a log of the output, but I don't see anything in the output that makes sense, relating to this issue.

I ran it as `steam -noverifyfiles |& tee $HOME/.steam.log`. If I need to mask out any info, let me know. Grepping for my passwords and IP doesn't show anything.

I did notice in the log that I'm in the 'publicbeta' client beta, if that helps any.

**Comment by EricHsuYC** [comment]
I am having this issue in my win64 laptop and school PC

**Comment by ronchaine** [comment]
This hit me as well when I installed steam to a new computer.

Mouse works, keyboard works, just the text input doesn't.  I can even copy-paste the textboxes, just can't write into them?  Maybe something to do with `SDL_TEXTINPUT`?  Backspace also works, as does copy-paste and undo with keyboard (as well as selecting the text with shift + arrows)

My setup is kinda weird though, since I run steam in a docker container, but maybe it makes debugging this easier since it should be reproducible pretty easily and seems to affect people other than me.

Dockerfile:
```
# change to steamos container if possible at some point
from archlinux:latest

# install base steam
RUN echo "[multilib]" >> /etc/pacman.conf && \
    echo "Include = /etc/pacman.d/mirrorlist" >> /etc/pacman.conf && \
    pacman -Syu --noconfirm \
    ca-certificates \
    zenity \
    wget \
    vulkan-radeon \
    lib32-vulkan-radeon \
    steam

# "Fix" locale warning
RUN localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF8

# Configure rest
RUN useradd -m steam \
    && usermod -a -G video,input,audio steam

user steam
cmd ["steam"]
```
And the script to run this:
```
#!/bin/sh
export DISPLAY=${DISPLAY:=:0}
docker run --rm \
    -v /dev/dri:/dev/dri \
    -v /tmp:/tmp \
    -v /dev/shm:/dev/shm \
    -v /etc/machine-id:/etc/machine-id \
    -v /container-data/steam/steam:/home/steam/.steam \
    -v /container-data/steam/share:/home/steam/.local/share/Steam \
    --privileged=true \
    --network=host \
    -e DISPLAY=${DISPLAY} \
    crx-steam
```

**Comment by Noxellar** [comment]
Aight, so i see that this issue is still open and I present you with this: https://steamcommunity.com/app/221410/discussions/2/558746089789084726/

I haven't seen anything else about this but one possible problem is that you haven't set an "input method framework" for this (I don't know how why that is even possible or an option).

Now I'm using Mint (Cinnamon) so I can't tell for sure whether this will apply for Arch but once I switched from "None" to literally any other option, it started working (for me, only steam overlay wasn't working, not the steam client itself). Now that I think about it, the problems did arise when I switched from Mate to Cinnamon and for some reason, Cinnamon didn't already have those preinstalled...

![Screenshot from 2021-12-30 13-53-52](https://user-images.githubusercontent.com/32978806/147718146-852cc0aa-7bda-4271-accc-862348eb86a4.png)
![Screenshot from 2021-12-30 13-54-19](https://user-images.githubusercontent.com/32978806/147718152-35697c95-8651-464d-9a35-12c85ce41422.png)

However, on closer inspection, it seems that you only need them installed. Before, I hadn't had any other input methods installed but once they were, input still worked regardless of whether it was set to "None" or not (even on restarting the game although I haven't tested this with a full restart of the computer).

TL;DR
You might not have an (active) input method. Try installing one and choosing that as the active input method.

**Comment by ErikReider** [comment]
@Noxellar This works perfectly! Thanks! :)

**Comment by Vaniog** [comment]
I solved this problem with installing font packages
`winetricks corefonts`

**Comment by wold9168** [comment]
The same problem. Steam doesn't respond anyway. And some games in steam have the same problem also.
The information of my computer:
```
OS: Arch Linux x86_64 
DE: Plasma 6.0.3 
IM: fcitx5-rime(wayland)
```

**Comment by Mrr7782** [comment]
Same as the previous comment. Can't type anywhere in Steam.
Steam stable
OS: Arch Linux (kernel 6.8.2)
DE: KDE Plasma 6.0.2 on X11 (xorg-server 21.1.11)
IM: fcitx (4.2.9.9)

**Comment by kisak-valve** [comment]
Hello @Mrr7782 update your system to libx11-1.8.8-3. There is an upstream regression at https://gitlab.freedesktop.org/xorg/lib/libx11/-/issues/205 which was triaged with that specific package build while we wait for 1.8.9.

**Comment by Mrr7782** [comment]
Updated `libx11` to `1.8.8-3`, fixed. Thank you and sorry.
